src/components/DateInput/YearAndMonthSelector.tsx

```diff
@@ -1,30 +1,54 @@
 import React from 'react';
-import PropTypes from 'prop-types';
+import type { LocaleUtils } from 'react-day-picker';
+
+// { date, localeUtils, onChange, startYear, endYear }
+export interface YearAndMonthSelectorProps {
+  /** Selected or current date as native JS `Date` object */
+  date: Date;
+
+  /** `react-day-picker` localeUtils */
+  localeUtils: LocaleUtils;
+
+  /** Year range start as number */
+  startYear: number;
+
+  /** Year range end as number */
+  endYear: number;
+
+  /** Month/year change callback */
+  onChange?: (date: Date) => void;
+}
 
 /**
  * Private component for DateInput that renders Month/Year select inputs.
  * @param {Object} props react props
  * @returns {ReactElement}
  */
-const YearAndMonthSelector = ({ date, localeUtils, onChange, startYear, endYear }) => {
-  const months = localeUtils.getMonths();
+const YearAndMonthSelector = ({
+  date,
+  localeUtils,
+  onChange,
+  startYear,
+  endYear,
+}: YearAndMonthSelectorProps): JSX.Element => {
+  const months: ReturnType<typeof localeUtils.getMonths> = localeUtils.getMonths();
 
   const years = Array.from({ length: endYear - startYear }, (_, i) => i + startYear);
 
-  const handleChange = ({
-    target: {
-      form: { year, month },
-    },
-  }) => {
-    onChange(new Date(year.value, month.value));
+  const handleMonthChange: React.ChangeEventHandler<HTMLSelectElement> = (event) => {
+    onChange?.(new Date(date.getFullYear(), Number(event.target.value)));
+  };
+
+  const handleYearChange: React.ChangeEventHandler<HTMLSelectElement> = (event) => {
+    onChange?.(new Date(Number(event.target.value), date.getMonth()));
   };
 
   return (
-    <form className="DayPicker-Caption border--bottom">
+    <div className="DayPicker-Caption border--bottom">
```

Comment:
Could I understand why this was using `form` before, and the idea behind changing this to a `div`?

Reply:
it was giving us trouble with types, it was getting complicated to type it as a form with `React.FormEvent` and then we would've had to dynamically get the values with something like `e.currentTarget.elements.namedItem('email')`, it would have to have more checks but that's where it was getting complex.

the only reason i found for using a form was to update the values getting them both at once from the form, the alternative for that was getting the value for `month` from the state when we were getting the value for `year` from the input, and viceversa.